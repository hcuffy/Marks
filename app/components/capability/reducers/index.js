import { capabilityHandlers } from './reducerHandlers'
import { reducerActionHandler } from '../../../reducers/reducerUtils.js'

const initialLoadState = {
	classDropdown: false,
	studentDropdown: false,
	questionDropdown: false,
	classroom: null,
	classroomId: null,
	studentName: null,
	studentId: null,
	answers: [],
	questions: []
}

export const applyCapabilityChanges = (state = initialLoadState, action) => {
	return reducerActionHandler(state, action, capabilityHandlers)
}
