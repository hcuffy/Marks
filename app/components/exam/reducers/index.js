import {examHandlers} from './reducerHandlers';
import {reducerActionHandler} from '../../../reducers/reducerUtils.js';

const initialLoadState = {
    subject:           '',
    openClassDropdown: false,
    openSubList:       false,
    classroomId:       null,
    selectedSubject:   null,
    examModal:         false,
    isInvalid:         false,
    isModalInvalid:    false,
    title:             '',
    weight:            ''
};

export function applyFilteredExam(state = initialLoadState, action) {
    return reducerActionHandler(state, action, examHandlers);
}
