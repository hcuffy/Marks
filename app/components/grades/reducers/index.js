import {gradeHandlers} from './reducerHandlers';
import {reducerActionHandler} from '../../../reducers/reducerUtils.js';

const initialLoadState = {
    classroomId:       null,
    subjectName:       null,
    classroomDropdown: false,
    subDrop:           false
};

export const applyGradeData = (state = initialLoadState, action) => {
    return reducerActionHandler(state, action, gradeHandlers);
};
