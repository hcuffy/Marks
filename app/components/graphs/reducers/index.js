import {graphHandlers} from './reducerHandlers';
import {reducerActionHandler} from '../../../reducers/reducerUtils.js';

const initialLoadState = {
    classroomId:       null,
    subjectName:       null,
    examName:          null,
    subjectId:         null,
    examId:            null,
    classroomDropdown: false,
    openSubList:       false,
    openExamList:      false,
    chartToDisplay:    null,
    chartTitle:        null
};

export const applyGraphData = (state = initialLoadState, action) => {
    return reducerActionHandler(state, action, graphHandlers);
};
