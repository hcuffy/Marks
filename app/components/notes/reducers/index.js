import {noteHandlers} from './reducerHandlers';
import {reducerActionHandler} from '../../../reducers/reducerUtils.js';

const initialLoadState = {
    studentDropdown: false,
    notesDropdown:   false,
    studentId:       null,
    noteId:          null,
    notes:           [],
    selectedStudent: null,
    selectedNote:    null,
    textBox:         null,
    textField:       null,
    isInvalid:       false
};

export const applyNotesData = (state = initialLoadState, action) => {
    return reducerActionHandler(state, action, noteHandlers);
};
