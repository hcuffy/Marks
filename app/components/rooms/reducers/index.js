export * from './classroom'
export * from './classroomModal'
export * from './tab'
