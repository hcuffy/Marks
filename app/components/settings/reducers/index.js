import {settingsHandlers} from './reducerHandlers';
import {reducerActionHandler} from '../../../reducers/reducerUtils.js';

const initialLoadState = {
    title:    null,
    street:   null,
    province: null,
    country:  null,
    zip:      null,
    city:     null,
    year:     null
};

const gradingLoadState = {note: true, points: false, percent: false};

export const applyGradeSystem = (state = gradingLoadState, action) => {
    return reducerActionHandler(state, action, settingsHandlers);
};

export const applyAddressData = (state = initialLoadState, action) => {
    return reducerActionHandler(state, action, settingsHandlers);
};
