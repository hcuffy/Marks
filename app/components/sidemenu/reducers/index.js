import {sidemenuHandlers} from './reducerHandlers';
import {reducerActionHandler} from '../../../reducers/reducerUtils.js';

const initialLoadState = {
    home:       '#1dbb90',
    classroom:  '',
    students:   '',
    exams:      '',
    graphs:     '',
    notes:      '',
    capability: '',
    settings:   ''
};

export const applyMenuStyling = (state = initialLoadState, action) => {
    return reducerActionHandler(state, action, sidemenuHandlers);
};
