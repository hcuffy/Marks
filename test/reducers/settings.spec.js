import { handleSchoolReducer, applyGradeSystem } from '../../app/reducers/settingsReducer'
import {
	DISPLAY_SCHOOL_DATA,
	HANDLE_SCHOOL_DATA,
	UPDATE_GRADING_DATA,
	GET_SYSTEM_TYPE
} from '../../app/constants/actionTypes'

const initialLoadState = {
	title: null,
	street: null,
	province: null,
	country: null,
	zip: null,
	year: null
}

const gradingLoadState = { note: true, points: false, percent: false }

describe('test settings reducer', () => {
	it('should use default address state', () => {
		expect(handleSchoolReducer(initialLoadState, {})).toMatchSnapshot()
	})

	it('should handle address creation', () => {
		const payload = {
			title: "Mark's School",
			street: 'Tester Street',
			province: 'Regensburg',
			country: 'Germany',
			zip: '93309',
			year: 2019
		}

		expect(
			handleSchoolReducer(initialLoadState, { type: HANDLE_SCHOOL_DATA, payload })
		).toMatchSnapshot()
	})

	it('should use default grading state', () => {
		expect(applyGradeSystem(gradingLoadState, {})).toMatchSnapshot()
	})

	it('should update grading system', () => {
		const payload = { note: false, points: true, percent: false }
		expect(
			applyGradeSystem(gradingLoadState, { type: UPDATE_GRADING_DATA, payload })
		).toMatchSnapshot()
	})

	it('should get grading system type', () => {
		const payload = { note: false, points: false, percent: true }
		expect(
			applyGradeSystem(gradingLoadState, { type: GET_SYSTEM_TYPE, payload })
		).toMatchSnapshot()
	})

	it('should use default displayed address state', () => {
		expect(applyGradeSystem(initialLoadState, {})).toMatchSnapshot()
	})

	it('should update displayed address', () => {
		const payload = {
			title: 'Test School',
			street: 'Tester Street',
			province: 'Kelheim',
			country: 'Germany',
			zip: '93309',
			year: 2018
		}

		expect(
			applyGradeSystem(initialLoadState, { type: DISPLAY_SCHOOL_DATA, payload })
		).toMatchSnapshot()
	})
})
