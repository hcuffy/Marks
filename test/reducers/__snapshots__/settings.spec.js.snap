// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test settings reducer should get grading system type 1`] = `
Object {
  "note": false,
  "percent": true,
  "points": false,
}
`;

exports[`test settings reducer should handle address creation 1`] = `
Object {
  "country": "Germany",
  "province": "Regensburg",
  "street": "Tester Street",
  "title": "Mark's School",
  "year": 2019,
  "zip": "93309",
}
`;

exports[`test settings reducer should update displayed address 1`] = `
Object {
  "country": null,
  "province": null,
  "street": null,
  "title": null,
  "year": null,
  "zip": null,
}
`;

exports[`test settings reducer should update grading system 1`] = `
Object {
  "note": false,
  "percent": false,
  "points": true,
}
`;

exports[`test settings reducer should use default address state 1`] = `
Object {
  "country": null,
  "province": null,
  "street": null,
  "title": null,
  "year": null,
  "zip": null,
}
`;

exports[`test settings reducer should use default displayed address state 1`] = `
Object {
  "country": null,
  "province": null,
  "street": null,
  "title": null,
  "year": null,
  "zip": null,
}
`;

exports[`test settings reducer should use default grading state 1`] = `
Object {
  "note": true,
  "percent": false,
  "points": false,
}
`;
